fusesoc/edatools/xsim.py

```diff
@@ -55,19 +133,39 @@ def build_main(self):
         #Build simulation model
         args = []
         args += [ self.toplevel]
+        args += [ 'glbl' ]
         args += ['--prj', 'xsim.prj']      # list of design files
         args += ['--timescale', '1ps/1ps'] # default timescale to prevent error if unspecified
         args += ['--snapshot', 'fusesoc']  # name of the design to simulate
         args += ['--debug', 'typical']     # capture waveforms
+        args += ['--relax']     # capture waveforms
+
+        #Get IP libraries
+        libs=[]
+        for src_file in src_files:
+            if src_file.file_type == 'xci':
+                ip_name = os.path.splitext(os.path.basename(src_file.name))[0]
+                ini_file = os.path.join(self.work_root, ip_name ,'xsim' , 'xsim.ini')
+                with open(ini_file, 'r') as f:
+                    for line in f:
+                        lib_name = line.split('=')[0]
+                        if lib_name not in libs:
+                            args += ['-L', lib_name]
+                            libs.append(lib_name)
+
+        args += ['-L', 'work']
+        args += ['-L', 'unisims_ver']
+        args += ['-L', 'unimacro_ver']
+        args += ['-L', 'secureip']
```

Comment:
These should probably be added to xsim_options instead

Reply:
Same here. This are common libraries that is included in Vivado generated scripts (tcl) when exporting IP. I was not aware of this libraries until I had to study generated sources. For user there is only way to figure out what libraries have to be specified --- analysis of all generated IP products. Maybe we should parse this tcl files the same way we parse ini.

Reply:
It's not enough with those libraries however. We also got simprims and XilinxCoreLib which might be needed. And then there's the equivalent libs for VHDL. I'm not sure how to best deal with this, but until we come up with a robust way I think it's better to have them in the .core files instead of pushing them to the FuseSoC code base